### ISSUE TITLE
Should we increase the line length to 100 or 120 characters in coding style?

### ISSUE THREAD

**Issue by Hzj-jie** [koreader/koreader#2180]
Currently in Coding style, the line length limitation is 80 characters. It's a little bit out of date, as we are using larger monitors now a days. And the most important thing is, the line length limitation is not really friendly for lambda driven language. A 'local function' has used 14 characters already.

What do you think?

**Comment by NiLuJe** [comment]
Yes please :).

I tend to be overly verbose in comments, and any decent editor should softwrap anyway, so, :D.

**Comment by houqp** [comment]
I can do 100. 120 is probably too long for horizontal split view in editor. Another disadvantage about long line is I find my eyes are not very good at horizontal scan, so long lines slows down my reading speed. Not sure if this is the same for other people though.

**Comment by NiLuJe** [comment]
Good point about split-view!

**Comment by Frenzie** [comment]
Perhaps I'm wrong to do so, but unlike e.g. variable naming and indentation, I see line length more as a guideline. Specifically, don't make your lines too long and 80 is a good rule of thumb as to what constitutes too long. But also don't go doing silly ASCII art things just because you happen to have a line of 90 or a 100, especially when you exceeded 80 only because of indentation. (Although technically, I suppose 80 + 20% margin means a "harder" limit of 96.) In any case, when you only slightly exceed 80 ASCII art is not yet conducive to readability, while hampering editability. Because those are our concerns, not rigid application of rules intended to improve readability and editability.

tl;dr 100 is fine.

@houqp Depends on the kind of 120. 120 - 4 indentations = 104, just to name an example. Of course if you're using too many indentations you might also want to think about restructuring things some way or other, but there's a difference between real 80/100/120 and a shorter line that just so happens to end up at column 80/100/120.

**Comment by houqp** [comment]
> @houqp Depends on the kind of 120. 120 - 4 indentations = 104, just to name an example. Of course if you're using too many indentations you might also want to think about restructuring things some way or other, but there's a difference between real 80/100/120 and a shorter line that just so happens to end up at column 80/100/120.

AFAIK, this is the main reason why 80 characters limit is enforced in Linux kernel: to discourage multi-level nested blocks and encourage rewriting long block of code into functions to increase readability. Although the 80 limit is somewhat arbitrary. It dates back to the time when terminal can only display 80 characters per line, which inherited from the line limit of punch card.

**Comment by Frenzie** [comment]
According to typographic wisdom something like 50-80 (or more usually 60-70
iirc) is the best line length, corroborated by research. It's certainly
possible that terminal line length isn't all that arbitrary but rather
optimal plus a small margin. ;)

**Comment by Frenzie** [comment]
Then again, punch cards wouldn't have typographic concerns. =p

**Comment by houqp** [comment]
that's interesting, maybe people actually used to read punch cards like novels! anyway, @chrox , you good with bumping limit to 100?

**Comment by houqp** [comment]
close this for now since we all agree on 100 char guidance.
